venues/aclweb.org/ACL/migrate_submissions_ARR_ACL.py

```diff
@@ -0,0 +1,776 @@
+            acl_metareview = openreview.Note(
+                forum = acl_blind_submission_forum,
+                replyto = acl_blind_submission_forum,
+                invitation = f'aclweb.org/ACL/2022/Conference/-/Meta_Review',
+                signatures = arr_metareview.signatures,
+                readers = [
+                    'aclweb.org/ACL/2022/Conference/Program_Chairs'
+                ],
+                writers = [
+                    'aclweb.org/ACL/2022/Conference'
+                ],
+                content = arr_metareview.content
+            )
+            acl_metareview_posted = client.post_note(acl_metareview)
+            assert acl_metareview_posted, print('failed to post metareview ', acl_metareview.id)
+            acl_metareviews_dictionary[acl_metareview_posted.signatures[0]] = acl_metareview_posted.replyto
+            
+# Retrieve all commitment submissions, ACL submissions, ACL blind submissions, and ACL blind submission reviews  
+commitment_notes = list(openreview.tools.iterget_notes(client,invitation='aclweb.org/ACL/2022/Conference/-/Commitment_Submission'))
+acl_submissions = list(openreview.tools.iterget_notes(client,invitation='aclweb.org/ACL/2022/Conference/-/Submission'))
+blind_submissions = {note.original: note for note in list(openreview.tools.iterget_notes(client, invitation = 'aclweb.org/ACL/2022/Conference/-/Blind_Submission'))}
+acl_reviews_dictionary = {review.signatures[0] : review.replyto for review in list(openreview.tools.iterget_notes(client, invitation = 'aclweb.org/ACL/2022/Conference/-/Official_Review'))}
+acl_metareviews_dictionary = {review.signatures[0] : review.replyto for review in list(openreview.tools.iterget_notes(client, invitation = 'aclweb.org/ACL/2022/Conference/-/Meta_Review'))}
+print(acl_reviews_dictionary)
+# Save all submissions in a dictionary by paper_link
+acl_submission_dict = {acl_submission.content['paper_link'].split('=')[1]:acl_submission for acl_submission in acl_submissions}
+
+for note in commitment_notes: 
+    # Get the arr submission forum. Then check if that is in the submission dictionary. If it is, check for blind. If not, run post_acl
+    arr_submission_forum = note.content['paper_link'].split('=')[1]
+    if arr_submission_forum not in acl_submission_dict:
+        print('link of commitment note not found in submission dict: ', note.id)
+        post_acl_submission(arr_submission_forum)
+    
+    # Check if the acl_submission.id is in blind_submissions
+    elif acl_submission_dict[arr_submission_forum].id not in blind_submissions:
+        print('acl submission for this commitment note not in blind: ', note.id)
+        post_blind_submission(acl_submission_dict[arr_submission_forum].id, acl_submission_dict[arr_submission_forum], client.get_note(arr_submission_forum))
+    
+    # Check if the blind submission has reviews
+    else: 
+        print('acl submission for this commitment note does not have reviews: ', note.id)
```

Comment:
this is misleading because you are not checking here if the submission already has reviews posted, any note that has an ACL submission and a blind copy would enter this else, right?

Reply:
Yes sorry that was more for my debugging purposes, I'll remove the print lines. It does not matter because once it enters post_reviews it checks for existing reviews anyway. Though I just realized that if a paper has reviews but not metareviews, it will not execute post_metareviews, so I need to change that

Reply:
It might be clearer to run both post_reviews() and post_metareviews() from here, instead of running one within the other